### DISCUSSION TITLE
What dose @cyanChill use to develop this app

### DISCUSSION THREAD

**Discussion by ghost** [MissingCore/Music#72] [General]
I want to know so i can make my own apps or contribute to this project like what software do you use etc also what do you use to compile it.

**Comment by cyanChill** [comment]
Since I come from a web development background, I've decided to use React Native as the framework to build out the app. We utilize Expo, which makes developing React Native apps easier to do. The tooling used is mainly what I'm familiar with and works on the web & on mobile.
- Here's a [blog](https://expo.dev/blog/from-web-to-native-with-react) on Expo's site about the difference between web code & mobile code.
- Expo also has a router that allows us to have the different screens seen (it wraps around `react-navigation`).
- Styling is done through Tailwind CSS via `nativewind`, which allows us to use Tailwind in React Native (since there's no concept of "class" for styling).
- State management is done with Jotai & React Query.
- We use SQlite for the database, interfacing through Drizzle ORM.

The most important thing is reading documentation or Google it when you get stuck on something.
- This is a [tutorial](https://docs.expo.dev/tutorial/introduction/) that Expo published to give you a general idea on what Expo does.
- There are different ways how you can test your project. [This page](https://docs.expo.dev/get-started/set-up-your-environment/?mode=development-build&buildEnv=local) explains the options; we use a development build because somethings won't work with the "Expo Go" option.
